contracts/script/multisigTransactionProposals/README.md

```diff
@@ -142,6 +142,55 @@ ts-node contracts/script/multisigTransactionProposals/safeSDK/delegateStake.ts $
 
 3. Follow the URL in the terminal output to the multisig wallet to sign the proposal and eventually execute it
 
+## Decoding Function Data
+
+### Prerequisites
+
+1. Node.js and npm: Ensure you have `Node.js` and `npm` installed on your system. You can download them from nodejs.org.
+2. Project Setup: Make sure your project is set up correctly with all necessary dependencies installed. Run
+   `npm install` in the project root to install dependencies.
+3. Compiled Contract ABIs: Ensure that the contract ABIs are compiled and available in the out directory by running
+   `forge build`. The script expects the ABI files to be located at out/<ContractName>.sol/<ContractName>.json. If this
+   doesn't work, start a nix shell, `nix-shell` and then run the build command
+
+### Usage
+
+To decode proposal data using the `decodeFunctionData` script, follow these steps:
+
+1. Assuming you're in the root folder
+2. Run the script using the following command and ensure that the $CONTRACT_NAME and $ENCODED_DATA are set
+
+```bash
+   ts-node contracts/script/multisigTransactionProposals/safeSDK/decodeFunctionData.ts $CONTRACT_NAME $ENCODED_DATA
+```
+
+Expected Output: The script will output the function name and arguments decoded from the provided data. If there is an
+error, such as an incorrect contract name or invalid encoded data, an error message will be displayed.
+
+### Example Usage
+
+Let's say you have a transaction proposal like this where some input data is encoded. This indicates that it's a
+function call that has been encoded when the proposal to `upgradeToAndCall(...)` was triggered. You can check the
+contract the proposal is interacting with for the $CONTRACT_NAME and copy the encoded data so that you can decode it
+using this ts-node tool.
+
+![Image of transaction proposal with encoded data](images/safeProposal.png)
+
+In this example, the command is:
+
+```bash
+ts-node contracts/script/multisigTransactionProposals/safeSDK/decodeFunctionData.ts LightClientV2 0x167ac6180000000000000000000000000000000000000000000000000000000000002710
```

Comment:
@alysiahuggins i get this error

```
ts-node contracts/script/multisigTransactionProposals/safeSDK/decodeFunctionData.ts LightClientV2 0x167ac6180000000000000000000000000000000000000000000000000000000000002710
No function call found in the encoded data
```
I did run `forge build`

Reply:
ah it's decoding a function call `updateEpochStartBlock(uint256)`, I have the abi built but this branch didn't have the updates to LightClientV2 so the abi generated from forge build wouldn't have included it. Perhaps pull and try now.